crates/sui-core/src/authority/execution_time_estimator.rs

```diff
@@ -62,6 +69,12 @@ pub struct ExecutionTimeObserver {
     // executing transactions. This is used to decide which observations should be shared
     // via consensus.
     object_utilization_tracker: LruCache<ObjectID, ObjectUtilization>,
+
+    sharing_rate_limiter: RateLimiter<
+        governor::state::NotKeyed,
+        governor::state::InMemoryState,
+        governor::clock::DefaultClock,
+    >,
```

Comment:
i wonder if we could use the RateLimiter to replace the leaky integrator? we wouldn't actually be limiting the rate of anything, just observing events and noting whether we are hitting the limit.

not terribly important, but it might let us delete some code